### DISCUSSION TITLE
How do I invoke a template within a file for preview while maintaining only the template?

### DISCUSSION THREAD

**Discussion by hollowhemlock** [SilentVoid13/Templater#470] [Help]
## Line in relevant file to be invoked in preview:

<%+ invoke the relevant template during preview mode %>

## Relevant Template:

templates/area.md

```
## Current Projects
```dataview
list
from "areas/<% tp.file.title %>" and #para/project and -#para/project/archived and -#type/area
sort created_at descending
\```
```

## Specifics
What goes in the relevant file where the template is to be used?

**Comment by AB1908** [comment]
Can you describe your use case?

**Reply by hollowhemlock** [reply]
I have a number of files that are formatted as areas. In each of these files there is a query using the dataview plugin to look for files that are linked to the areas.

One day I open up an area file and I don't like the query. I would like to edit one file (like a class that takes inputs) and update the presentation across all the area files instead of having to edit the template and reinsert the template in all the area files individually.

**Reply by AB1908** [reply]
You may want to look at dataview's `dv.view` API. I don't think there's a clean way to replace across several files with templater.

**Reply by AB1908** [reply]
I mean, it can be done, but it wouldn't be the right solution IMO. The templater way can get messy.

**Comment by hollowhemlock** [comment]
Thanks so much for pointing me to dv.view. I’ll play around with that. Link
to documents for others interested:
https://blacksmithgu.github.io/obsidian-dataview/api/code-reference/
